grid.py

```diff
@@ -23,139 +19,99 @@ def __init__(self, num_rows, num_cols, tile_length):
-            ang_deg = i[0]
-            ang_rad = ang_deg * math.pi / 180
-            distance = i[1]
-            if distance != -1:
-                x_obst = distance * math.cos(ang_rad)
-                y_obst = distance * math.sin(ang_rad)  # upper left origin
-                col = self._get_idx(x + x_obst, False)
-                row = self._get_idx(y + y_obst, True)
-                if (not col == None and not row == None):
-                    if (self.grid[row][col] in pathSet):
-                        returner = True
-                    self.grid[row][col].isFound = True
-                    self.grid[row][col].is_obstacle = True
-                    self.grid[row][col].is_bloated = False
-                    if (self.bloat_tile(row, col, radius, bloat_factor, pathSet) == True):
-                        returner = True
-        return returner
+        objs, non_objs = self.sensor_data_to_tiles(tup_data, x, y, sensor_type)
+        #print(non_objs)
+        for non_obj in non_objs:
+            #print("hiiiiiii")
+            before = non_obj.is_obstacle
+            non_obj.decrease_score(sensor_type)
+            if before == True and non_obj.is_obstacle == False:
+                print("debloating")
+                self.debloat_tile(non_obj)
 
-    def bloat_tile(self, row, col, radius, bloat_factor, pathSet=set()):
-        """
-        Bloats tiles in grid around the obstacle with index [row][col] within radius [radius].
-        Going off grid, could final tile get bloated?
-        TODO EDGE CASES
-        """
-        bloat_radius = radius * bloat_factor
-        index_radius_inner = int(bloat_radius / self.tileLength) + 1
-        index_rad_outer = index_radius_inner + 2
-
-        lower_row = int(max(0, row - index_rad_outer))
-        lower_col = int(max(0, col - index_rad_outer))
-        upper_row = int(min(row + index_rad_outer, self.num_rows))
-        upper_col = int(min(col + index_rad_outer, self.num_cols))
         returner = False
-        for i in range(lower_row, upper_row):
-            for j in range(lower_col, upper_col):
-                curr_tile = self.grid[i][j]
-                y_dist = abs(i - row)
-                x_dist = abs(j - col)
-                dist = math.sqrt(x_dist * x_dist + y_dist * y_dist)
-                # print("dist: " + str(dist))
-                if (dist < index_radius_inner):
-                    if (not curr_tile.is_obstacle):
-                        curr_tile.is_obstacle = True
-                        curr_tile.is_bloated = True
-                        if (curr_tile in pathSet):
-                            returner = True
+        for obj in objs:
+            if obj in path_set:
+                returner = True
+            obj.is_found = True
+            obj.is_bloated = False
+            obj.increase_score(sensor_type)
+            if self.bloat_tile(obj, radius, bloat_factor, path_set):
+                returner = True
         return returner
 
+
```

Comment:
Might be useful to go back and update the specification more precisely

Reply:
Yes please

Reply:
Resolving